featuretools/tests/entityset_tests/test_es.py

```diff
@@ -1199,10 +1200,14 @@ def _check_schema_version(version, es, warning_text):
         'relationships': relationships,
     }
 
-    if warning_text:
-        with pytest.warns(UserWarning) as record:
-            deserialize.description_to_entityset(dictionary)
-        assert record[0].message.args[0] == warning_text
+    # TODO Refactor this function and both parent functions to run tests with a less complex implementation
+    earlier_schema_warning = 'is no longer supported by this version'
+    if earlier_schema_warning in str(warning_text):
+        logger = logging.getLogger('featuretools')
+        logger.propagate = True
+        deserialize.description_to_entityset(dictionary)
+        assert warning_text in caplog.text
+        logger.propagate = False
```

Comment:
Before we were checking that a `UserWarning` got raised for both earlier and later schema versions with the assert statement inside the `pytest.warns` context manager. Now we changed one of these to be logged instead.

With these code changes here, we are only testing that warning is logged for the earlier schema test. We are no longer testing that the `UserWarning` still gets raised for the later schema test. To test that we need to add a later scheme condition to the if-else block to verify the warning gets raised as it did before.

Reply:
One way you could refactor this would be to split `_check_schema_version` into three different functions. You could do something like `_check_schema_raises_warning` to check that the `UserWarning` gets raised inside `test_later_schema_version`. Then have `_check_schema_logs_warning` to test logging for the `test_earlier_schema_version` test. In both of those new utility functions we could use a third `_get_schema_dictionary` function to build and return the dictionary that will be needed by both of the new check functions.

Another idea would be to create a new parameter that gets passed to _check_schema_version to indicate whether we should check for a `UserWarning` or a logged warning instead of relying on the value of the warning text to make that determination.

This seems like it is getting more complicated than I anticipated though. @rwedge Do you have any thoughts on the best approach here?

Reply:
I think keep `_check_schema_version` as a single function, add a new parameter to it to indicate if it is expected to warn / log / neither, and use `warning_text` for matching purposes